src/deploymentstatus.h

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2016-2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BITCOIN_DEPLOYMENTSTATUS_H
+#define BITCOIN_DEPLOYMENTSTATUS_H
+
+#include <chain.h>
+#include <sync.h>
+#include <versionbits.h>
+
+#include <limits>
+
+extern RecursiveMutex cs_main;
+
+/** Global cache for versionbits deployment status */
+extern VersionBitsCache versionbitscache GUARDED_BY(cs_main);
+
+/** Determine if a deployment is active for the next block */
+inline bool DeploymentActiveAfter(const CBlockIndex* pindexPrev, const Consensus::Params& params, Consensus::BuriedDeployment dep)
```

Comment:
This dispatch scares me a bit unless we use enum class -- enums could coerce poorly right?

Reply:
Nope; int (and the like) won't automatically coerce to enum, so with `int x = Consensus::DEPLOYMENT_SEGWIT; return DeploymentActiveAt(pindex, params, x)` you get errors:

```
validation.cpp:1921:12: error: no matching function for call to 'DeploymentActiveAt'
    return DeploymentActiveAt(pindex, consensusparams, x);
           ^~~~~~~~~~~~~~~~~~
./deploymentstatus.h:32:13: note: candidate function not viable: no known conversion from 'int' to 'Consensus::BuriedDeployment' for 3rd argument
inline bool DeploymentActiveAt(const CBlockIndex* pindex, const Consensus::Params& params, Consensus::BuriedDeployment dep)
            ^
./deploymentstatus.h:37:13: note: candidate function not viable: no known conversion from 'int' to 'Consensus::DeploymentPos' for 3rd argument
inline bool DeploymentActiveAt(const CBlockIndex* pindex, const Consensus::Params& params, Consensus::DeploymentPos dep) EXCLUSIVE_LOCKS_REQUIRED(cs_main)
            ^
```

(even if it did, which enum it coerced to is ambiguous between the two types, resulting in a failure like `error: call to 'DeploymentActiveAt' is ambiguous`)